src/components/database/Queries.js

```diff
@@ -0,0 +1,274 @@
+        email: user.email,
+        isAdmin: user.isAdmin,
+        joinDate: user.joinDate,
+        name: user.name,
+        needsAttention: user.needsAttention,
+        partnerEmail: user.partnerEmail,
+        count: user.count,
+      }: UserType);
+    case 'pending-members':
+      return ({
+        count: user.count,
+        email: user.email,
+        date: user.date,
+        partnerEmail: user.partnerEmail,
+      }: PendingType);
+    case 'actions':
+      return ({
+        count: user.count,
+        date: user.date,
+        message: user.message,
+      }: ActionType);
+    case 'solved-actions':
+      return ({
+        count: user.count,
+        date: user.date,
+        message: user.message,
+      }: ActionType);
+    default:
+      throw Error('The collection you asked for doesn\'t exist.');
+  }
+}
+
+function actionObject(count: number, user: UserType,
+    message: string): ActionType {
+  const actionMessage = `[${message}]: ${user.email}`;
+  return ({
+    date: fieldValue.serverTimestamp(),
+    count: count,
+    message: actionMessage,
+  });
+}
+
+function newUserObject(count: number, user: PendingType): UserType {
+  return ({
+    adminNote: '',
+    email: user.email,
+    isAdmin: false,
+    joinDate: fieldValue.serverTimestamp(),
+    name: '',
+    needsAttention: false,
+    partnerEmail: user.partnerEmail,
+    count: count,
+  }: UserType);
+}
+/**
+  * fieldQuery returns a number of documents ordered by a field,
+  * @param {string} collection specifies the collection in which we query
+  * @param {string} orderField the field we order by
+  * @param {any} start first document in the order chosen
+  * @param {number} limit the number of documents to be retrieved
+  * @return {Array<UserType | PendingType | ActionType>} an array of the objects
+  * created from the documents
+  */
+async function fieldQuery(collection: string, orderField: string, start: any,
+    limit: number): Promise<Array<UserType | PendingType | ActionType>> {
+  const collectionRef = database.collection(collection);
+
+  try {
+    const data = await collectionRef
+        .orderBy(orderField)
+        .startAt(start)
+        .limit(limit)
+        .get();
+
+    return data.docs
+        .map((doc: any): any => doc.data())
+        .map((doc: any): UserType | PendingType | ActionType =>
+          sanitize(collection, doc));
+  } catch (error) {
+    console.log(error);
+    return [];
+  }
+}
+
+/**
+  * findDocumentQuery returns documents that match some value of a field
+  * @param {string} collection specifies the collection in which we query
+  * @param {string} field specifies the field we query by
+  * @param {any} value specifies the value of the field
+  * @return {Array<UserType | PendingType | ActionType>} an array of objects
+  * created from the documents
+ */
+async function findDocumentQuery(collection: string, field: string,
+    value: string | number | boolean): Promise<Array<UserType | ActionType |
```

Comment:
So the return type of this function is Promise, or Array?

Reply:
it is marked as `async`, so it returns a Promise from whatever returned.

Reply:
That is correct but the signature in comments suggests an Array and this is confusing to me.

> @return {Array<UserType | PendingType | ActionType>} an array of objects
    * created from the documents`

Reply:
the problem here is that Flow will consider any type of `async/await` function to be returning a promise even if the failure of the promise is handled inside the block of the `try/catch` function. This means that in order to not get a flow error, I must declare that my function returns a promise, even if in reality it will always return an array.